### DISCUSSION TITLE
API-Portal Upgrade to V2 Breaks Documenations

### DISCUSSION THREAD

**Discussion by dallas-fletchall** [mia-platform/community#575] [Q&A] [answered]
### What product are you having troubles with?

Console

### What Console version are you using?

13.2.0

### Description

After upgrading the API-Portal as described in the [migration docs](https://docs.mia-platform.eu/docs/runtime_suite/api-portal/configuration#how-to-migrate-to-v2), I am no longer able to excerise endpoints on the `documentations/api-portal` endpoint.

Old versions:
- `api-portal:1.16.10` -> `api-portal:2.1.0`
- `swagger-aggregator:3.5.1` -> `swagger-aggregator:3.7.0`

### Actual Outcome

New API-Portal w/o ability to use `Try-it out` functionality of SwaggerUI.

![new-api-portal](https://github.com/user-attachments/assets/dfa15267-9b1d-4c03-aa83-47ecdabeb0c6)

### Expected Outcome

New API-Portal, but no host information and broken `Try-it out` functionality of SwaggerUI. (Image below is old API-Portal, but working host).

![old-api-portal](https://github.com/user-attachments/assets/18f0f07f-7e78-4bc8-ab2e-4cff327cbcc1)

**Comment by LiliumCandidum** [comment]
Hi @dallas-fletchall ,

We need your open api definition to help you better.
You can find it by:
1. Open the API Portal with the console inspector open
2. Go to the network tab and look for the `api/openapi/json?subswaggersInDescription=false` call. It is listed as `json?subswaggersInDescription=false`
3. Click it and copy what's inside the "response" tab
4. Send here what you copied (pay attention to any confidential information you don't want to share publicly)

And also, did you edit the default swagger-aggregator configuration? If yes, please send it to us

**Comment by dallas-fletchall** [comment]
Thanks for the quick response. 

Our  swagger-aggregator.json file is below
```json
{
  "baseSwagger": {
    "swagger": "2.0",
    "consumes": ["application/json"],
    "produces": ["application/json"],
    "securityDefinitions": {
      "oidc": {
        "type": "oauth2",
        "flows": {
          "authorizationCode": {
            "authorizationUrl": "https://example.com/oauth2/authorize",
            "tokenUrl": "https://example.com/oauth2/token",
            "scopes": {
              "openid": "openId scope"
            }
          }
        }
      }
    }
  }
}
```

After looking at how long that subswaggers document was, and being unsure what we want to disclose about our API, I looked into it a little harder. It seems that the `servers` variable is causing our issue.

The original servers object was:
```json
"servers": [
    {"url":"http://tag-creation-webhook-consumer" },
    {"url":"http://version-approval-api"}
]
```

This led to the issues that we observed above. By changing the url of one of the servers, we got the functionality for that portion back.
```json
"servers": [
    {"url":"/"},
    {"url":"http://version-approval-api"}
]
```

**Can you confirm this is the issue?**
**Do we have alternative options that do not require container updates?**

**Reply by LiliumCandidum** [reply, marked as answer]
Yes, that was the issue.

You could add them in your `swagger-aggregator.json` configuration, inside the `baseSwagger` object but it's not possible yet to visualize the list on the API Portal and thus not usable. We already planned it and if you need it we can let you know when it's done.

However, if your list of servers is inside a specific path, there is nothing you can do because the "nested" servers list overrides the "global" one.

**Comment by dallas-fletchall** [comment]
Thanks for all the help. Please keep me informed if this gets updated.

**Reply by LiliumCandidum** [reply]
Hi @dallas-fletchall,

I just want you to know that we released a new API Portal `v2.2.0` that supports the global servers list. I also suggest to update the Swagger Aggregator to the latest version.

Have a nice day,
Paola
